parlai/tasks/coco_caption/agents.py

```diff
@@ -102,19 +136,36 @@ def __getitem__(self, index):
         else:
             image_id = self.test_info['images'][index]['id']
         ep = {
-            'text': self.dict_agent.txt2vec(QUESTION),
+            'text': QUESTION,
             'image': self.get_image(image_id),
             'episode_done': True,
         }
+        if self.no_intro:
+            ep.pop('text')
         if self.opt.get('extract_image', False):
             ep['image_id'] = image_id
             return ep
         if not self.datatype.startswith('test'):
             anno = self.annotation['annotations'][index]
             ep['labels'] = [anno['caption']]
-            ep['valid'] = True
+            if self.datatype.startswith('train'):
+                candidates = random.Random(index).choices(self.cands, k=150)
```

Comment:
it looks like here you are supplying valid cands while training?

Reply:
Similar to my earlier comment to Alex, we are sending labels through with validation cands during training. However, we are not using them except during validation. The loss we are using (`ContrastiveLoss`) only uses the truth labels for the batch.

Reply:
ok, do we need to send them through if we are not using them?